FileWriteFlags.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+namespace FileWriteFlag
+{
+	// Default values are Do not overwrite files, append to existing files, and write in binary format.
+	enum FileWriteFlag : unsigned int
```

Comment:
Consider using an enum class instead of the namespace wrapper

Reply:
If I use enum class, then I have to perform static_cast in order to do bitwise operations:

Write(FileStreamWriter(filename, static_cast<unsigned int>(FileWriteFlag::Overwrite) | static_cast<unsigned int>(FileWriteFlag::Truncate)), mapData);

Associated error if static_cast is missing: E0349 No operator "|" matches these operands.

This seems too unwieldy to me. If I treat them as an older enum but wrap them in a namespace, I don't have to perform the static_cast but I'm also not placing each member of the enum in the global namespace.

Why are you saying to consider a enum class here?

Reply:
Hmm, good point. I hadn't thought of that.

There may be a workaround though:
http://blog.bitwigglers.org/using-enum-classes-as-type-safe-bitmasks/